2024-06-18 09:46:35,879: asyncio: INFO: [MLTrader] Executing the initialize lifecycle method
2024-06-18 09:46:37,204: root: INFO: Current backtesting datetime 2023-12-15 08:30:00-05:00
2024-06-18 09:46:37,205: asyncio: INFO: [MLTrader] Executing the before_market_opens lifecycle method
2024-06-18 09:46:37,206: root: INFO: Current backtesting datetime 2023-12-15 09:30:00-05:00
2024-06-18 09:46:37,206: asyncio: INFO: [MLTrader] Executing the before_starting_trading lifecycle method
2024-06-18 09:46:37,207: asyncio: INFO: [MLTrader] [34mExecuting the on_trading_iteration lifecycle method at 2024-06-18 09:46:37[0m
2024-06-18 09:46:37,207: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-06-18 09:46:37,965: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-06-18 09:46:38,088: asyncio: ERROR: [MLTrader] 'NewsV2' object has no attribute '_dict_'
2024-06-18 09:46:38,091: asyncio: ERROR: [MLTrader] Traceback (most recent call last):
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 923, in run
    self._run_trading_session()
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 871, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 262, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 286, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 397, in _on_trading_iteration
    raise e
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\strategies\strategy_executor.py", line 369, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\cordo\.vscode\pythonCode\tradingBot.py", line 52, in on_trading_iteration
    news = self.get_news()
           ^^^^^^^^^^^^^^^
  File "C:\Users\cordo\.vscode\pythonCode\tradingBot.py", line 44, in get_news
    news = [ev._dict_["_raw"]["headline"]for ev in news]
            ^^^^^^^^^
  File "C:\Users\cordo\AppData\Local\Programs\Python\Python312\Lib\site-packages\alpaca_trade_api\entity.py", line 29, in __getattr__
    return super().__getattribute__(key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NewsV2' object has no attribute '_dict_'. Did you mean: '__dict__'?

2024-06-18 09:46:38,094: asyncio: INFO: [MLTrader] Executing the on_bot_crash event method
2024-06-18 09:46:38,104: root: INFO: Backtesting finished
2024-06-18 09:46:38,448: asyncio: WARNING: [MLTrader] Cannot plot returns because the strategy returns are missing
2024-06-18 09:46:38,594: asyncio: WARNING: [MLTrader] Cannot create a tearsheet because the strategy returns are missing
2024-06-18 09:46:38,594: backtest_stats: INFO: Backtest took 0:00:02.717111 for a speed of 0.000
